### PR TITLE
feat(ocap-kernel): integrate Snaps attenuated endowment factories

### LINKED ISSUE #935: Integrate Snaps attenuated endowment factories into vat globals
## Context

In #813 / #933 we made the vat global allowlist configurable and expanded it with host/Web API endowments. The current implementation passes raw `globalThis` references (e.g. `globalThis.setTimeout`, `globalThis.Date`) as endowments.

MetaMask Snaps has battle-tested endowment factories that provide **attenuated** versions of these globals with security and lifecycle improvements. We opened [MetaMask/snaps#3957](https://github.com/MetaMask/snaps/pull/3957) to export them via `@metamask/snaps-execution-environments/endowments`.

## What to do

Once `snaps#3957` is merged and released, update `packages/ocap-kernel/src/vats/endowments.ts` to import and use the Snaps factories:

### Replace with attenuated versions

- **`setTimeout` / `clearTimeout`** → `timeout.factory()` — handle isolation (vats can't clear each other's timers), type checking, teardown function for vat termination
- **`Date`** → `date.factory()` — `Date.now()` adds ~1ms noise (anti-timing-attack, monotonic)

### Add new endowments

- **`setInterval` / `clearInterval`** → `interval.factory()` — same attenuation as timeout
- **`crypto` / `SubtleCrypto`** → `crypto.factory()` — hardened Web Crypto API
- **`Math`** → `math.factory()` — replaces `Math.random()` with crypto-backed RNG

### Integrate teardown

The `timeout` and `interval` factories return a `teardownFunction`. Wire these into `VatSupervisor.terminate()` to clean up pending timers when a vat is stopped.

### Keep as-is

`TextEncoder`, `TextDecoder`, `URL`, `URLSearchParams`, `atob`, `btoa`, `AbortController`, `AbortSignal` — Snaps' versions just do `harden(X)` which is effectively what we already do.

## Location

- `packages/ocap-kernel/src/vats/endowments.ts`
- `packages/ocap-kernel/src/vats/VatSupervisor.ts` (terminate method)

### PR DESCRIPTION
**Pull request by sirtimid**
## Summary

Part 2 of the vat endowments overhaul (closes #935 once merged). Replaces raw vat-global endowments with attenuated factories from `@metamask/snaps-execution-environments/endowments`:

- **Added** `setInterval`, `clearInterval`, `crypto`, `SubtleCrypto`, and `Math` (attenuated `Math.random` sourced from `crypto.getRandomValues` — not a CSPRNG per the upstream NOTE, but defends against stock-RNG timing side channels) to the default allowlist.
- **Replaced** raw `setTimeout`/`clearTimeout` and `Date` with the monotonic, attenuated Snaps factories — `Date.now()` is now monotonically clamped with random jitter (repeated calls within the same ms return the same value).
- **Wired teardown**: each factory's `teardownFunction` is aggregated and invoked from `VatSupervisor.terminate()` via `Promise.allSettled`, so a single failing teardown doesn't mask others. Teardown errors are logged and never block kernel-stream closure.
- **API change**: `DEFAULT_ALLOWED_GLOBALS` export + `VatSupervisor`'s `allowedGlobals` ctor param are replaced with `createDefaultEndowments()` factory + `makeAllowedGlobals` ctor param. The factory is invoked once per supervisor so each vat gets isolated state (vats cannot clear each other's timers).

Follow-up network-endowment integration tracked separately in #936.

## Dependencies

Depends on [MetaMask/snaps#3957](https://github.com/MetaMask/snaps/pull/3957), which adds the `/endowments` subpath export. That has shipped — pinned at `@metamask/snaps-execution-environments@^11.1.0`, which resolves to the released `11.1.0`.

## Follow-up test fixes in this PR

Two unit tests broke on the post-lockdown test environment after the integration and are fixed in a follow-up commit:

- `endowments.test.ts > teardown cancels pending timers` — `vi.useFakeTimers()` tries to assign `Date.now` on `ClockDate extends NativeDate`, but `lockdown(overrideTaming: 'severe')` makes that slot non-writable. Replaced with a real 30 ms delay (same pattern as `remote-comms.test.ts:584`).
- `VatSupervisor.test.ts > calls the endowments teardown before closing the stream` — `vi.spyOn(stream, 'end')` fails because `BaseDuplexStream` hardens the instance. Threaded `TestDuplexStream`'s existing `writerOnEnd` callback through the test helper and used it as the ordering proxy.

Also includes a drive-by fix for a flaky e2e test (`remote-comms.test.ts > handles reconnection with exponential backoff`) using the same `ackTimeoutMs: 5_000` pattern as PR #927.

## Test plan

- [x] `yarn install` succeeds
- [x] `yarn workspace @metamask/ocap-kernel test:dev:quiet` passes — 2276/2276 tests — covers `endowments.test.ts` (shape, isolation, teardown-cancels-timers) and `VatSupervisor.test.ts` (teardown-before-close, teardown-throw-still-closes-stream, double-terminate safety, factory-invoked-once)
- [x] `yarn workspace @metamask/kernel-node-runtime test:e2e:ci remote-comms.test.ts` passes — 21/21 e2e tests
- [x] `yarn workspace @ocap/kernel-test test:dev:quiet` passes — 23/23 files, 92 passed + 3 todo — covers integration tests for `setInterval`/`clearInterval`, `crypto.getRandomValues`, `Math.random`, and tamed-throws cases
- [x] `yarn lint` clean
- [x] `yarn build` clean — 24/24 tasks successful

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **High Risk**
> High risk because it changes SES vat endowment behavior and the public API for configuring globals (`DEFAULT_ALLOWED_GLOBALS`/`allowedGlobals` -> `createDefaultEndowments`/`makeAllowedGlobals`), which can break consumers and affects sandboxed execution semantics.
> 
> **Overview**
> Integrates Snaps’ attenuated endowment factories into vat globals via a new `createDefaultEndowments()` factory, expanding defaults to include `setInterval`/`clearInterval`, `crypto`/`SubtleCrypto`, and `Math` (crypto-backed `Math.random`), and switching timers/`Date` to Snaps’ attenuated implementations.
> 
> Replaces the exported `DEFAULT_ALLOWED_GLOBALS` constant and `VatSupervisor`’s `allowedGlobals` ctor param with a per-supervisor `makeAllowedGlobals` factory that returns `{ globals, teardown }`, and updates `VatSupervisor.terminate()` to run endowment teardown (aggregating/logging failures) before closing the kernel stream.
> 
> Updates docs and tests to describe/request endowments via `VatConfig.globals`/`Kernel.make({ allowedGlobalNames })`, adds coverage for new globals and restriction edge-cases, and tweaks a remote-comms e2e test to use a longer `ackTimeoutMs` to reduce flakiness.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 589a89a941d28afc13ad692b02c211a8e40d774e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/ocap-kernel/src/vats/endowments.ts [hunk 1]
```diff
@@ -1,31 +1,138 @@
...
+      const failures = results
+        .filter(
+          (result): result is PromiseRejectedResult =>
+            result.status === 'rejected',
+        )
+        .map((result) => result.reason);
+      if (failures.length > 0) {
+        throw new AggregateError(
+          failures,
+          `Endowment teardown failed (${failures.length}/${teardowns.length})`,
```
Comment (rekmarks) [inline packages/ocap-kernel/src/vats/endowments.ts]:
```suggestion
          `One or more endowment teardowns failed (${failures.length}/${teardowns.length})`,
```

**Review by rekmarks — DISMISSED** [review]
LGTM, with one nit

**Review by rekmarks — APPROVED** [review]
